### PR TITLE
Update Lifecycle to 2.8.5

### PR DESCRIPTION
**Pull request by terrakok**
Copied from 7cb2ab613631f03ea788409214d76bcc876e85bb

### CODE REVIEW

File: lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt [hunk 1]
```diff
@@ -119,7 +119,7 @@ fun <T : Any> SavedStateHandle.saveable(
     init: () -> T,
 ): PropertyDelegateProvider<Any?, ReadOnlyProperty<Any?, T>> =
     PropertyDelegateProvider { thisRef, property ->
-        val classNamePrefix = if (thisRef != null) thisRef::class.canonicalName + "." else ""
```
Comment (igordmn) [inline lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt]: This should be `canonicalName`. During `jb-main` merge to `integration-releaser/lifecycle/...` there should be a conflict, in which `canonicalName` should be chosen (I should did it myself in the past when merged 2.8.4, sorry 🫠)

Comment (igordmn) [reply lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt]: Also, please, push `integration-release/lifecycle/2.8`

Comment (terrakok) [reply lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt]: done

**Review by MatkovIvan — APPROVED** [review]

File: lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt (hunk 1 above)
Comment (igordmn) [reply lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt]: After merging the PR, also merge `jb-main` again to `integration-release/lifecycle/2.8`

**Review by igordmn — APPROVED** [review]

**Comment by igordmn** [comment]
Removed "Release Notes", as we don't mention version bumps separately (they are in "Dependencies" section

**Pull request merged**